esmvalcore/preprocessor/_dask_progress.py

```diff
@@ -0,0 +1,236 @@
+    ) -> None:
+        self._desc = f"{description} " if description else description
+        self._log_interval = dask.utils.parse_timedelta(
+            log_interval, default="s"
+        )
+        self._prev_elapsed = 0.0
+        interval = dask.utils.parse_timedelta("1s", default="s")
+        super().__init__(dt=interval)
+
+    def _draw_bar(self, frac: float, elapsed: float) -> None:
+        if (elapsed - self._prev_elapsed) < self._log_interval and frac != 1.0:
+            return
+        self._prev_elapsed = elapsed
+        pbar = "#" * int(self._width * frac)
+        percent = int(100 * frac)
+        elapsed_fmt = dask.utils.format_time(elapsed)
+        desc_width = 30
+        msg = (
+            f"{self._desc:<{desc_width}}[{pbar:<{self._width}}] | "
+            f"{percent:3}% Completed | {elapsed_fmt}"
+        )
+        logger.info(msg)
+
+
+class DistributedProgressLogger(
+    distributed.diagnostics.progressbar.TextProgressBar
+):
+    """Progress logger for the Dask distributed scheduler."""
+
+    # Disable warnings about design choices that have been made in the base class.
+    # pylint: disable=too-few-public-methods,unused-argument,useless-suppression
+
+    def __init__(
+        self,
+        keys,
+        log_interval: str | float = "1s",
+        description: str = "",
+    ) -> None:
+        self._desc = f"{description} " if description else description
+        self._log_interval = dask.utils.parse_timedelta(
+            log_interval, default="s"
+        )
+        self._prev_elapsed = 0.0
+        super().__init__(keys, interval="1s")
+
+    def _draw_bar(
+        self,
+        remaining: int,
+        all: int,  # pylint: disable=redefined-builtin
+        **kwargs,
+    ) -> None:
+        frac = (1 - remaining / all) if all else 1.0
+        if (
+            self.elapsed - self._prev_elapsed
+        ) < self._log_interval and frac != 1.0:
+            return
+        self._prev_elapsed = self.elapsed
+        pbar = "#" * int(self.width * frac)
+        percent = int(100 * frac)
+        elapsed = dask.utils.format_time(self.elapsed)
+        desc_width = 30
+        msg = (
+            f"{self._desc:<{desc_width}}[{pbar:<{self.width}}] | "
+            f"{percent:3}% Completed | {elapsed}"
+        )
+        logger.info(msg)
+
+    def _draw_stop(self, **kwargs):
+        pass
+
+
+def _compute_with_progress(
+    delayeds: Iterable[Delayed],
+    description: str,
+) -> None:
+    """Compute delayeds while displaying a progress bar."""
+    use_distributed = True
+    try:
+        distributed.get_client()
+    except ValueError:
+        use_distributed = False
+
+    log_progress_interval = dask.utils.parse_timedelta(
+        CFG["logging"]["log_progress_interval"],
+        default="s",
+    )
+    if CFG["max_parallel_tasks"] != 1 and log_progress_interval == 0.0:
+        # Enable progress logging if `max_parallel_tasks` > 1 to avoid clutter.
+        log_progress_interval = 1.0
```

Comment:
```suggestion
        log_progress_interval = 10.0
```

I think the default should be more than 1s, otherwise this will fill the logs of complex recipes very quickly.